### ISSUE TITLE
Interactive Bokeh control and ipywidegets issue

### ISSUE THREAD

**Issue by cmccarthy1** [KxSystems/jupyterq#46]
The following issue was raised by Seth Oldham in K4 Listbox, this issue presented here is to allow for up to date progress to be reported.

I have put together a simple notebook with an interactive Bokeh control using the following code
```
## /%python 

from ipywidgets import interact
import numpy as np

from bokeh.io import push_notebook, show, output_notebook
from bokeh.plotting import figure
from bokeh.resources import INLINE

x = np.linspace(0, 2*np.pi, 2000)
y = np.sin(x)

p = figure(title="simple line example", plot_height=300, plot_width=600, y_range=(-5,5),
           background_fill_color='#efefef')
r = p.line(x, y, color="#8888cc", line_width=1.5, alpha=0.8)

def update(f, w=1, A=1, phi=0):
    if   f == "sin": func = np.sin
    elif f == "cos": func = np.cos
    r.data_source.data['y'] = A * func(w * x + phi)
    push_notebook()

output_notebook(resources=INLINE)

show(p, notebook_handle=True);
interact(update, f=["sin", "cos"], w=(0,50), A=(1,10), phi=(0, 20, 0.1));
```

When I run that with the Python kernel, the interactive controls work as expected. But when I switch to the kdb kernel, and uncomment the /%python magic, the plot displays as expected but the controls don't have any effect. The logging from the notebook server does not reveal anything that seems useful, but I am including it here in case I'm missing something

```
jupyter-notebook[71056]: [W 09:43:29.914 NotebookApp] Trusting notebook /seth_dev/sample_bokeh.ipynb
jupyter-notebook[71056]: [I 09:43:30.194 NotebookApp] Starting buffering for 9a2cccee-2e5f-4652-a71e-664b7b914d62:1531d9414a2f49dda92e0e0c35b81ff2
jupyter-notebook[71056]: [W 09:43:30.994 NotebookApp] 404 GET /nbextensions/qtconsole/main.js?v=20190531123200 () 3.31ms referer=http:///notebooks/seth_dev/sample_bokeh.ipynb
jupyter-notebook[71056]: [W 09:43:31.120 NotebookApp] 404 GET /nbextensions/widgets/notebook/js/extension.js?v=20190531123200 (192.168.48.120) 2.22ms referer=http:///notebooks/seth_dev/sample_bokeh.ipynb
jupyter-notebook[71056]: [I 09:43:32.543 NotebookApp] Adapting to protocol v5.1 for kernel 9a2cccee-2e5f-4652-a71e-664b7b914d62
jupyter-notebook[71056]: [I 09:45:32.697 NotebookApp] Saving file at /seth_dev/sample_bokeh.ipynb
jupyter-notebook[71056]: [I 09:47:49.860 NotebookApp] Starting buffering for 9a2cccee-2e5f-4652-a71e-664b7b914d62:4eb97bf98a164d4785c8a1a44bf0cb4f
jupyter-notebook[71056]: [I 09:47:49.970 NotebookApp] Kernel shutdown: 9a2cccee-2e5f-4652-a71e-664b7b914d62
jupyter-notebook[71056]: [I 09:47:50.011 NotebookApp] Kernel started: 570a550e-a314-4b2d-978a-2d1462d0cf41
jupyter-notebook[71056]: [I 09:47:51.444 NotebookApp] Adapting to protocol v5.1 for kernel 570a550e-a314-4b2d-978a-2d1462d0cf41
jupyter-notebook[71056]: [I 09:49:32.655 NotebookApp] Saving file at /seth_dev/sample_bokeh.ipynb
```

**Comment by Dianeod** [comment]
Issue fixed in PR https://github.com/KxSystems/jupyterq/pull/56.

**Comment by Dianeod** [comment]
Issue fixed in PR https://github.com/KxSystems/jupyterq/pull/56
Also note in the above example that 
```
r.data_source.data['y'] = A * func(w * x + phi)
```
Must be amended to
```
r.data_source.data['y'] = list(A * func(w * x + phi))
```

**Comment by cmccarthy1** [comment]
Issue fixed in PR #56

**Comment by cmccarthy1** [comment]
Issue fixed in PR #56
